src/xenia/kernel/xam/xam_user.cc

```diff
@@ -600,17 +601,42 @@ dword_result_t XamUserCreateTitlesPlayedEnumerator_entry(
       !handle_ptr) {
     return X_ERROR_INVALID_PARAMETER;
   }
-  if (!kernel_state()->xam_state()->IsUserSignedIn(user_index)) {
-    return X_ERROR_INVALID_PARAMETER;
+
+  const uint32_t kEntrySize = sizeof(XTitleEnumerator::XTITLE_PLAYED);
+  if (buffer_size_ptr) {
+    *buffer_size_ptr = kEntrySize * game_count;
   }
 
-  // auto e = new XStaticEnumerator<X_XDBF_GPD_TITLEPLAYED>(kernel_state(),
-  // game_count); auto result = e->Initialize(user_index, 0xFB, 0xB0050,
-  // 0xB000B, 0x20, game_count, 0);
+  auto e = object_ref<XTitleEnumerator>(
+      new XTitleEnumerator(kernel_state(), game_count));
+  auto result =
+      e->Initialize(user_index, 0xFB, 0xB0050, 0xB000B, 0x20, game_count, 0);
+  if (XFAILED(result)) {
+    return result;
+  }
+
+  const auto user = kernel_state()->xam_state()->GetUserProfile(user_index);
```

Comment:
Move this check before creation of enumerator. There is no need for enumerator to exist if there is no user to use it